### ISSUE TITLE
[BUG] @example sections not formatted correctly

### ISSUE THREAD

**Issue by petemcwilliams** [compodoc/compodoc#304]
##### **Overview of the issue**

@example code is incorrectly formatted. Html tags are displaying, `&quot;` and other characters have been translated and output.

```
    /**
     * This method will display a dialog to the user to capture some arbitary text, you
     * supply the buttons that should be displayed with the callback that should be invoked
     * for each (if required).
     *
     * @param {string} title the title for the dialog
     * @param {string} message the messsage to be presented to the user
     * @param {*} buttonConfigs an array, containing a buttonName and callback, i.e.
     * @param {boolean} [enableBackdropDismiss=true]  optional parameter - allow user to dismiss the prompt by clicking
     *                                                on the backdrop (default to true)?
     *
     * @example
     * ```typescript
     * this.alertCtrl.showTextInputWithButtons(
     *  "Pin Verification",
     *  "You have been sent an email. Please check your email and then type the pin code that you have received below.",
     *  [
     *      { buttonName: "Cancel", callback: null },
     *      { buttonName: "Verify", callback: (value: string) => { this.sendPin(value); } }
     *  ]);
     * ```
     *
     * @memberOf IonicAlerterService
     */
    public abstract showTextInputWithButtons(
        title: string,
        message: string,
        buttonConfigs: any,
        enableBackdropDismiss?: boolean,
    ): void;
```
version: 1.0.1
![screen shot 2017-09-08 at 16 43 29](https://user-images.githubusercontent.com/28921238/30219726-e28e139e-94b4-11e7-8e3f-101d53452d65.png)

Has changed post version 1.0.0-beta.13 example below is as expected

![screen shot 2017-09-08 at 16 41 49](https://user-images.githubusercontent.com/28921238/30219662-ad262462-94b4-11e7-97db-3f04c8bebfd8.png)

##### **Operating System, Node.js, npm, compodoc version(s)**

    Node : v8.2.1
    npm  : 5.3.0 
    OS   : macOS Sierra

    "@compodoc/compodoc": "1.0.1"

##### **Angular configuration, a `package.json` file in the root folder**

        "@angular/animations": "4.1.3",
        "@angular/common": "4.1.3",
        "@angular/compiler": "4.1.3",
        "@angular/compiler-cli": "4.1.3",
        "@angular/core": "4.1.3",
        "@angular/forms": "4.1.3",
        "@angular/http": "4.1.3",
        "@angular/platform-browser": "4.1.3",
        "@angular/platform-browser-dynamic": "4.1.3",
        "@angular/platform-server": "4.1.3",
        "@compodoc/compodoc": "1.0.1",
        "@types/jasmine": "2.5.41",
        "@types/node": "7.0.5",
        "codelyzer": "^3.1.2",
        "ionic-angular": "3.6.0",
        "ncp": "^2.0.0",
        "rexreplace": "^2.5.2",
        "rimraf": "^2.6.1",
        "rxjs": "5.4.0",
        "ts-node": "2.0.0",
        "tslint": "^5.6.0",
        "tslint-consistent-codestyle": "^1.6.0",
        "tslint-eslint-rules": "^4.1.1",
        "tslint-ionic-rules": "^0.0.11",
        "typescript": "2.3.4",
        "zone.js": "0.8.12"

##### **Compodoc installed globally or locally ?**

locally

##### **Motivation for or Use Case**

See diagrams

##### **Reproduce the error**

Example above

##### **Related issues**

There was some changes to @example between 1.0.0-beta.13 and 1.0.0-beta.14

**Comment by stewwan** [comment]
Needing this solution here, is there a workaround?

**Comment by stewwan** [comment]
I figured out downgrade to beta version 12 seems to fix it

**Comment by mfdeveloper** [comment]
Hey @stewwan and @petemcwilliams My workaround is remove the `@example` jsdoc tag, and move the example code to description, like this:

```ts
/**
 * My documentated class
 *
 * ## Example:
 *
 * ```ts
 * \@Component({
 *  selector: 'minha-pagina'
 *  templateUrl: 'minha-pagina.html'
 * })
 * class MyComponent implements OnInit  {

 * }
 * ```
 * @class FormItemBasePage
 * @implements {OnInit}
 */
export class MyComponent {
}

```

**Obs:** Preppend a "\\" when you use **decorators** (`@Component, @Input...`) in your examples. Whether you don't do this, the tag `<pre>` is broken. Seems be a parser markdown problem.

**Comment by stewwan** [comment]
Nice thanks @mfdeveloper

**Comment by krotscheck** [comment]
This doesn't seem to resolve the issue. Markup is:

    /**
     * ```ts
     * \@NgModule({
     *    imports: [
     *      ServiceModule,
     *    ],
     *    ...
     * })
     * ```
     */

And it renders like this: https://imgur.com/AoVHS03

**Comment by mfdeveloper** [comment]
Yeah @krotscheck you are correct! My suggestions is only a "workaround" for render the highlight example code snippet. Sadly, the "\" is rendered to 😞 

I think that @vogloblinsky will correct this soon, right?

**Comment by vogloblinsky** [comment]
This issue is related to TypeScript parsing JSDoc comment and @NgModule decorator. Interference with @xxx JSDoc tags.
I have to try this deeper with TypeScript sources and submit a PR.

**Comment by FabienDehopre** [comment]
What do you think about this npm package https://www.npmjs.com/package/jsdoc-escape-at ?

**Comment by miller45** [comment]
Is there a workaround to add like a preprocessor to replace @example with #example? Can we add this preprocessor to typescript or typedoc?
We do not want mess around with our repo to replace all example tags

**Comment by WmDevelopment** [comment]
Any chance to solve the bug on the next release?

**Comment by vogloblinsky** [comment]
Fixed in 1.1.14
